load.go

```diff
@@ -383,3 +384,12 @@ func isNotFoundErr(err error) bool {
 	var notFound *viper.ConfigFileNotFoundError
 	return err != nil && errors.As(err, &notFound)
 }
+
+// skipField determines whether to skip a field
+// when using reflection to process the application's config
+// structure. Skip fields that are unexported, unless
+// they are anonymous, since anonymous fields are embedded fields,
+// which must be processed in case they contain exported members.
+func skipField(f reflect.StructField) bool {
+	return !f.IsExported() && !f.Anonymous
```

Comment:
did we come to a conclusion about if we needed to consider if this is a struct or pointer to a struct?

Reply:
I'll add a unit test that covers the case where the embedded struct is embedded as a pointer and see if any further changes are needed.

Reply:
@wagoodman I think this may be limitation in mapstructure we're running into.

I made a gist to illustrate what I mean: https://gist.github.com/willmurphyscode/9e9e65b7c090252dcb2e261fdd55b29b

Reply:
It turns out it works if the embedded pointer points to an exported type. I've updated the docs to point out this restriction and added a unit test for this case.